### PR TITLE
Trustar Connector Changes

### PR DESCRIPTION
**Pull request by sathya573**
Trustar Connector Changes

### CODE REVIEW

File: stix_shifter_modules/trustar/entry_point.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        
+        # ...implement your own setup similar to the following:
+        
+        basepath = os.path.dirname(__file__)
+        filepath = os.path.abspath(os.path.join(basepath, "stix_translation"))
+
+        dialect = 'default'
+        query_translator = QueryTranslator(options, dialect, filepath)
+        results_translator = JSONToStix(options, dialect, filepath)
+        self.add_dialect(dialect, query_translator=query_translator, results_translator=results_translator, default=True)
```
[the remark this answers is not in the document]
Comment (mdazam1942) [reply stix_shifter_modules/trustar/entry_point.py:39]: I usually generate recent requirements.py by running `python3 generate_requirements.py`
Then create virtual env by running this command:
`virtualenv -p python3 virtualenv && source virtualenv/bin/activate && pip install -r requirements-dev.txt`

**Comment by mdazam1942** [comment]
@cookna let me know if you are still working on this PR. do you want us to continue the review?

File: stix_shifter_modules/trustar/stix_transmission/api_client.py [hunk 2]
```diff
@@ -0,0 +1,376 @@
+from stix_shifter_utils.stix_transmission.utils.RestApiClient import RestApiClient
+from stix_shifter_utils.utils.error_response import ErrorResponder
+from .error_mapper import ErrorMapper
+import requests
+import requests.auth
```
Comment (mdazam1942) [inline stix_shifter_modules/trustar/stix_transmission/api_client.py:5]: not sure why do you import requests library here?

File: stix_shifter_modules/trustar/stix_transmission/api_client.py [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+        response = self.get_token()
+        if response['code'] == 200:
+            self.token = json.loads(response['data'])['access_token']
+        else:
+            return response
+        
+        self.headers["Authorization"] = "Bearer {}".format(self.token)
+
+        ping_result = self.ping_trustar() 
+        return ping_result
```
[the remark this answers is not in the document]
Comment (mdazam1942) [reply stix_shifter_modules/trustar/stix_transmission/api_client.py]: @cookna this is still not fixed.

File: stix_shifter_modules/trustar/stix_transmission/api_client.py [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+        response = self.get_token()
+        return response
+
+    def get_search_results(self, search_id, range_start=None, range_end=None):
+        # Return the search results. Results must be in JSON format before being translated into STIX
+        response = {}
+        try:
+            response = self._get_results(search_id, range_start, range_end)
+        except Exception as e:
+            response = {'success':False, 'code': 2000, 'data': []}
```
Comment (mdazam1942) [inline stix_shifter_modules/trustar/stix_transmission/api_client.py:35]: this response doesn't make sense to the user. specially code `'code': 2000`. I would suggest you to look at other existing modules for exception handlings.

File: stix_shifter_modules/trustar/stix_translation/json/from_stix_map.json [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      "x_searchTerm": ["searchTerm"],
+      "x_type": ["entityTypes"],
+      "x_report_details": ["reportId"],
+      "x_report_ids": ["reportIds"],
+      "x_enclaveIds" : ["enclaveIds"],
+      "x_tags": ["tags"],
+      "x_excludeTags": ["x_excludeTags"]
+    }
+  },
+  "x-bitcoin": {
```
Comment (mdazam1942) [inline stix_shifter_modules/trustar/stix_translation/json/from_stix_map.json:13]: follow the same naming conventions as `x-trustar-query` for all custom objects such as `x-trustar-bitcoin`

Comment (mdazam1942) [reply stix_shifter_modules/trustar/stix_translation/json/from_stix_map.json:13]: also you need to follow the same naming conventions in to_stix_map.

File: stix_shifter_modules/trustar/stix_translation/json/from_stix_map.json [hunk 6]
```diff
@@ -0,0 +1,70 @@
+{
+  "x-trustar-query": {
+    "fields": {
+      "x_searchTerm": ["searchTerm"],
```
Comment (mdazam1942) [inline stix_shifter_modules/trustar/stix_translation/json/from_stix_map.json:4]: you don't need `x_` prefix for these properties name

File: stix_shifter_modules/trustar/stix_translation/json/from_stix_map.json [hunk 7]
```diff
@@ -0,0 +1,70 @@
+{
+  "x-trustar-query": {
```
Comment (mdazam1942) [inline stix_shifter_modules/trustar/stix_translation/json/from_stix_map.json:2]: there's an inconsistency between the field mappings. in to_stix_map same fields are mapped to `x-trustar-report` object but here you are calling it `x-trustar-query`. I guess this object should be `x-trustar-query` as well.

have a look at other objects to make them consistent between two mapping files.

**Review by mdazam1942 — CHANGES_REQUESTED** [review]
please go through my comments and let me know if you have any questions

File: stix_shifter_modules/trustar/configuration/config.json [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        "report_limit": {
+            "type": "number",
+            "optional": true,
+            "default": 300
+        },
+        "max_number_pages": {
+            "type": "number",
+            "optional": true,
+            "default": 10
+        },
```
[the remark this answers is not in the document]
Comment (deepsiIBM) [reply stix_shifter_modules/trustar/configuration/config.json:26]: Implemented as suggested review it and merge it.

File: stix_shifter_modules/trustar/stix_translation/json/from_stix_map.json [hunk 9]
```diff
@@ -0,0 +1,70 @@
+{
+  "x-trustar": {
```
[the remark this answers is not in the document]
Comment (deepsiIBM) [reply stix_shifter_modules/trustar/stix_translation/json/from_stix_map.json]: @mdazam1942
 Implemented as suggested review it and merge it.

File: stix_shifter_modules/trustar/stix_transmission/api_client.py [hunk 10]
```diff
@@ -0,0 +1,374 @@
...
+        self.headers["Authorization"] = "Bearer {}".format(self.token)
+
+        ping_result = self.ping_trustar() 
+        return ping_result 
+
+    def get_search_results(self, search_id, range_start=None, range_end=None):
+        # Return the search results. Results must be in JSON format before being translated into STIX
+        try:
+            resp = self._get_results(search_id, range_start, range_end)
+            response = {'success':True, 'code': 200, 'data': resp}
```
[the remark this answers is not in the document]
Comment (deepsiIBM) [reply stix_shifter_modules/trustar/stix_transmission/api_client.py]: @mdazam1942
Implemented as suggested review it and merge it.

File: stix_shifter_modules/trustar/stix_transmission/api_client.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (deepsiIBM) [reply stix_shifter_modules/trustar/stix_transmission/api_client.py]: @mdazam1942
Implemented as suggested review it and merge it.

File: stix_shifter_modules/trustar/stix_transmission/ping_connector.py [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+class PingConnector(BasePingConnector):
+    def __init__(self, api_client):
+        self.api_client = api_client
+        self.logger = logger.set_logger(__name__)
+
+    def ping_connection(self):
+        try:
+            response = self.api_client.ping_data_source()
+            print("PING RESPONSE: {}".format(response))
```
[the remark this answers is not in the document]
Comment (deepsiIBM) [reply stix_shifter_modules/trustar/stix_transmission/ping_connector.py]: @mdazam1942
ping_connection not needed so omitted this method.Please review and merge the change.

File: stix_shifter_modules/trustar/stix_translation/query_constructor.py [hunk 12]
```diff
@@ -0,0 +1,283 @@
...
+        FROM = now - range_t
+        TO = now
+
+    trustar_query_translator.return_obj["from"] = FROM
+    trustar_query_translator.return_obj["to"] = TO
+
+    ret = json.dumps(trustar_query_translator.return_obj)
+    ret = ret.replace('%', ' ')
+
+
```
[the remark this answers is not in the document]
Comment (deepsiIBM) [reply stix_shifter_modules/trustar/stix_translation/query_constructor.py]: @mdazam1942
Implemented as suggested review it and merge it.

File: stix_shifter_modules/trustar/configuration/config.json [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
```
[the remark this answers is not in the document]
Comment (deepsiIBM) [reply stix_shifter_modules/trustar/configuration/config.json:26]: @mdazam1942
Implemented as suggested review it and merge it.

File: stix_shifter_modules/trustar/stix_transmission/api_client.py (hunk 2 above)
Comment (deepsiIBM) [reply stix_shifter_modules/trustar/stix_transmission/api_client.py]: @mdazam1942
Implemented as suggested review it and merge it.

File: stix_shifter_modules/trustar/stix_transmission/api_client.py (hunk 4 above)
Comment (deepsiIBM) [reply stix_shifter_modules/trustar/stix_transmission/api_client.py]: @mdazam1942
Implemented as suggested review it and merge it.

File: stix_shifter_modules/trustar/stix_translation/json/from_stix_map.json (hunk 5 above)
Comment (deepsiIBM) [reply stix_shifter_modules/trustar/stix_translation/json/from_stix_map.json]: @mdazam1942
Implemented as suggested review it and merge it.

File: stix_shifter_modules/trustar/stix_translation/json/from_stix_map.json (hunk 6 above)
Comment (deepsiIBM) [reply stix_shifter_modules/trustar/stix_translation/json/from_stix_map.json]: @mdazam1942
Implemented as suggested review it and merge it.

File: stix_shifter_modules/trustar/stix_translation/json/from_stix_map.json (hunk 7 above)
Comment (deepsiIBM) [reply stix_shifter_modules/trustar/stix_translation/json/from_stix_map.json]: We have implemented like : x-trustar-report

File: stix_shifter_modules/trustar/stix_translation/json/to_stix_map.json [hunk 14]
```diff
@@ -0,0 +1,252 @@
...
+      "object": "file"
+    },
+    {
+      "key": "software.id",
+      "object": "software",
+      "references": "file"
+    }
+  ],
+  "firstSeen": {
+    "key": "x-trustar-indicator.x_firstseen",
```
[the remark this answers is not in the document]
Comment (deepsiIBM) [reply stix_shifter_modules/trustar/stix_translation/json/to_stix_map.json]: @mdazam1942
Implemented as suggested review it and merge it.

**Pull request closed without merging**